mongodbatlas/resource_mongodbatlas_project.go

```diff
@@ -322,10 +387,39 @@ func resourceMongoDBAtlasProjectRead(ctx context.Context, d *schema.ResourceData
 	return nil
 }
 
+func fetchUserDefinedLimits(definedLimitsList []*projectLimit, connV2 *admin.APIClient, ctx context.Context, projectID string) ([]admin.DataFederationLimit, diag.Diagnostics) {
+	definedLimitsMap := make(map[string]*projectLimit)
+	for _, definedLimit := range definedLimitsList {
+		definedLimitsMap[definedLimit.name] = definedLimit
+	}
+
+	fetchedLimits, _, err := connV2.ProjectsApi.ListProjectLimits(ctx, projectID).Execute()
+	if err != nil {
+		return nil, diag.Errorf("error getting project's limits (%s): %s", projectID, err)
+	}
+	filteredLimits := []admin.DataFederationLimit{}
+	for i := range fetchedLimits {
+		limitRes := fetchedLimits[i]
+		if _, ok := definedLimitsMap[limitRes.Name]; ok {
+			filteredLimits = append(filteredLimits, limitRes)
+		}
+	}
+	return filteredLimits, nil
+}
+
 func resourceMongoDBAtlasProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	updateProject(ctx, d, meta)
-	updateTeams(ctx, d, meta)
-	updateAPIKeys(ctx, d, meta)
```

Comment:
when implementing update for limits, identified that errors in these methods were not being handled.